lib/ansible/modules/network/basics/uri.py

```diff
@@ -403,8 +403,12 @@ def main():
 
     # Grab all the http headers. Need this hack since passing multi-values is
     # currently a bit ugly. (e.g. headers='{"Content-Type":"application/json"}')
+    warnings = set()
     for key, value in six.iteritems(module.params):
         if key.startswith("HEADER_"):
+            warnings.add('Supplying headers via HEADER_* is deprecated and '
```

Comment:
i was just going to mirror the current module.warnings .. module.deprecated and consume it in callback with display.deprecated vs display.warnings